### ISSUE TITLE
Quoting service (in persistence mode) having storage issues for unicode characters

### ISSUE THREAD

**Issue by elnyry-sam-k** [mojaloop/project#2471]
**Summary**:
Quoting service (in persistence mode) having storage issue with several unicode characters (and is failing to store several unicode characters).. For example here's an error
```
"errorInformation": {
        "errorCode": "2001",
        "errorDescription": "Internal server error - insert into `party` (`dateOfBirth`, `firstName`, `lastName`, `middleName`, `quotePartyId`) values ('1990-01-01', 'ကောင်းထက်စံ', 'ဒေါ်သန္တာထွန်', 'အောင်', 122) - ER_TRUNCATED_WRONG_VALUE_FOR_FIELD: Incorrect string value: '\\xE1\\x80\\x80\\xE1\\x80\\xB1...' for column 'firstName' at row 1"
      }
```

**Severity**:
Medium

**Priority**:
High

**Expected Behavior**
Quoting service DB has no issues storing all allowed unicode characters (allowed by the Name regex)

**Steps to Reproduce**
  1. Configure a party with unicode characters (example Burmese characters) such as ကောင်းထက်စံ'
  2. Send a quote request with the above party involved
  3. Observe results / callbacks / logs

**Specifications**
  - Component (if known): quoting service (DB)
  - Version: https://github.com/mojaloop/quoting-service/releases/tag/v12.0.10
  - Platform: AWS (but applies to all)
  - Subsystem: configuration of mysql DB in quoting service
  - Type of testing: Manual - RC testing (new test case addition for unicode characters)
  - Bug found/raised by: @mdebarros, @elnyry-sam-k 

**Notes**:

- Severity when opened: Medium
- Priority when opened: High

Notes:

More logs
```
ER_TRUNCATED_WRONG_VALUE_FOR_FIELD

(node:25) UnhandledPromiseRejectionWarning: FSPIOPError: insert into `party` (`dateOfBirth`, `firstName`, `lastName`, `middleName`, `quotePartyId`) values ('1990-01-01', 'ကောင်းထက်စံ', 'ဒေါ်သန္တာထွန်', 'အောင်', 122) - ER_TRUNCATED_WRONG_VALUE_FOR_FIELD: Incorrect string value: '\xE1\x80\x80\xE1\x80\xB1...' for column 'firstName' at row 1
at createFSPIOPError (/opt/quoting-service/node_modules/@mojaloop/central-services-error-handling/src/factory.js:198:12)
at Object.reformatFSPIOPError (/opt/quoting-service/node_modules/@mojaloop/central-services-error-handling/src/factory.js:333:12)
at CachedDatabase.createParty (/opt/quoting-service/src/data/database.js:634:34)
at runMicrotasks (<anonymous>)
at processTicksAndRejections (internal/process/task_queues.js:97:5)
Error: ER_TRUNCATED_WRONG_VALUE_FOR_FIELD: Incorrect string value: '\xE1\x80\x80\xE1\x80\xB1...' for column 'firstName' at row 1
at Query.Sequence._packetToError (/opt/quoting-service/node_modules/mysql/lib/protocol/sequences/Sequence.js:47:14)
at Query.ErrorPacket (/opt/quoting-service/node_modules/mysql/lib/protocol/sequences/Query.js:79:18)
at Protocol._parsePacket (/opt/quoting-service/node_modules/mysql/lib/protocol/Protocol.js:291:23)
at Parser._parsePacket (/opt/quoting-service/node_modules/mysql/lib/protocol/Parser.js:433:10)
at Parser.write (/opt/quoting-service/node_modules/mysql/lib/protocol/Parser.js:43:10)
at Protocol.write (/opt/quoting-service/node_modules/mysql/lib/protocol/Protocol.js:38:16)
at Socket.<anonymous> (/opt/quoting-service/node_modules/mysql/lib/Connection.js:88:28)
at Socket.<anonymous> (/opt/quoting-service/node_modules/mysql/lib/Connection.js:526:10)
at Socket.emit (events.js:311:20)
at addChunk (_stream_readable.js:294:12)
--------------------
at Protocol._enqueue (/opt/quoting-service/node_modules/mysql/lib/protocol/Protocol.js:144:48)
at Connection.query (/opt/quoting-service/node_modules/mysql/lib/Connection.js:198:25)
at /opt/quoting-service/node_modules/knex/lib/dialects/mysql/index.js:126:18
at new Promise (<anonymous>)
at Client_MySQL._query (/opt/quoting-service/node_modules/knex/lib/dialects/mysql/index.js:120:12)
at executeQuery (/opt/quoting-service/node_modules/knex/lib/execution/internal/query-executioner.js:37:17)
at Client_MySQL.query (/opt/quoting-service/node_modules/knex/lib/client.js:134:12)
at /opt/quoting-service/node_modules/knex/lib/execution/transaction.js:363:24
at new Promise (<anonymous>)
at Client_MySQL.trxClient.query (/opt/quoting-service/node_modules/knex/lib/execution/transaction.js:358:12)
(node:25) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). To terminate the node process on unhandled promise rejection, use the CLI flag `--unhandled-rejections=strict` (see https://nodejs.org/api/cli.html#cli_unhandled_rejections_mode). (rejection id: 35)
2021-09-10T15:19:21.177Z - info: 2021-09-10T15:19:21.177Z, (1631287161002:moja2-quoting-service-fdb7fb8b4-hmbnr:25:kt4drof7:91781) [quotesmodel]: Error occurred while handling error. Check service logs as this error may not have been propagated successfully to any other party: FSPIOPError: network error in sendErrorCallback: Request failed with status code 400
at Object.createFSPIOPError [as CreateFSPIOPError] (/opt/quoting-service/node_modules/@mojaloop/central-services-error-handling/src/factory.js:198:12)
at QuotesModel.sendErrorCallback (/opt/quoting-service/src/model/quotes.js:959:28)
at runMicrotasks (<anonymous>)
at processTicksAndRejections (internal/process/task_queues.js:97:5)

at Client_MySQL.trxClient.query (/opt/quoting-service/node_modules/knex/lib/execution/transaction.js:358:12) {code: 'ER_TRUNCATED_WRONG_VALUE_FOR_FIELD', errno: 1366, sqlMessage: 'Incorrect string value: '\xE1\x80\x80\xE1\x80\xB1...' for column 'firstName' at row 1', sqlState: 'HY000', index: 0, …}

17:49:59	insert into `party` (`dateOfBirth`, `firstName`, `lastName`, `middleName`, `quotePartyId`) values ('1987-01-01', 'က', 'dddd', 'dddd', 126)	Error Code: 1366. Incorrect string value: '\xE1\x80\x80' for column 'firstName' at row 1	0.205 sec
```

**Comment by mdebarros** [comment]
The MySQL default character-set is `latin1` which is not compatible with Unicode characters. It is recommended that MySQL servers are properly configured to utilise `utf8mb4` character set.

It is also possible to upgrade existing MySQL server installations to utilise `utf8mb4`, however, it seems to be less risky to instead upgrade to `utf8mb3` (`utf8` being an alias).

One can fix the issue by either upgrading the entire Database and tables to `utf8` encoding.

Alternatively as a quick fix...one can upgrade `party` table used for storing Party information by the Quoting Service as follows:

```sql
ALTER TABLE party CONVERT TO CHARACTER 
       SET utf8 COLLATE utf8_unicode_ci;
```

Upgrading to `utf8mb4` can similarly be done, but additional steps may be required to update index's and modify columns to cater for the additional bytes required by the encoding scheme.

**Comment by elnyry-sam-k** [comment]
will be closed with #2382

**Comment by elnyry-sam-k** [comment]
TTK results with v13..0.2 RC: https://mojaloop.slack.com/archives/GKBDSALSZ/p1631534704018500
